File: frontend/src/services/refundService.ts
import { apiService as api } from './api';

export interface Refund {
  id?: number;
  bill?: number;
  amount: number;
  reason: string;
  processedBy: string;
  processedAt?: string;
  status?: string;
}

export const processRefund = (billId: number, amount: number, reason: string, processedBy: string) =>
  api.post(`/billing/refunds?billId=${billId}&amount=${amount}&reason=${encodeURIComponent(reason)}&processedBy=${encodeURIComponent(processedBy)}`);

export const getRefundsByBill = (billId: number) =>
  api.get(`/billing/refunds/by-bill/${billId}`);

export const getAllRefunds = () =>
  api.get('/billing/refunds');

File: src/main/java/com/task/hms/billing/service/RefundService.java
package com.task.hms.billing.service;

import com.task.hms.billing.model.Refund;
import java.util.List;

public interface RefundService {
    Refund processRefund(Long billId, Double amount, String reason, String processedBy);
    List<Refund> getRefundsByBill(Long billId);
    List<Refund> getAllRefunds();
}
